src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -156,45 +156,50 @@ class CradleMessageExtractor(
         val buffer: MutableList<StoredMessage> = ArrayList()
         val remaining: LinkedList<StoredMessage> = LinkedList()
         while (iterator.hasNext()) {
-            sink.canceled?.apply {
-                logger.info { "canceled because: $message" }
-                return
-            }
-            prev = currentBatch
-            currentBatch = iterator.next()
-            check(prev.firstTimestamp <= currentBatch.firstTimestamp) {
-                "Unordered batches received: ${prev.toShortInfo()} and ${currentBatch.toShortInfo()}"
-            }
-            val needFiltration = prev.isNeedFiltration()
-
-            if (prev.lastTimestamp < currentBatch.firstTimestamp) {
-                if (needFiltration) {
-                    prev.messages.filterTo(buffer, StoredMessage::inRange and parameters.preFilter)
-                } else {
-                    prev.messages.preFilterTo(buffer)
+            val measurement = dataMeasurement.start("process_cradle_group_batch")
+            try {
```

Comment:
You can kotlin try-with-resources analog
```kotlin
            dataMeasurement.start("process_cradle_group_batch").use {
```

Reply:
in this case smart cast doesn't work 
Smart cast to 'StoredGroupedMessageBatch' is impossible, because 'prev' is a local variable that is captured by a changing closure